### ISSUE TITLE
getting genes associated with PANTHER terms

### ISSUE THREAD

**Issue by ktyssowski** [moosa-r/rbioapi#3]
thanks for this helpful tool! after running rba_panther_enrich(), how can I get the list of genes in my gene list that are associated with each significant term? I can't seem to find a function that will allow that. Thanks!

**Comment by moosa-r** [comment]
Dear Tyssowski 

Unfortunately, this PANTHER API endpoint doesn't return the user's gene within each significant term. However, In case you are performing an over-representation analysis of your genes against GO terms, I think you have two options here: 
 
1. In rbioapi, use other enrichment services. Please see this article: 
https://rbioapi.moosa-r.com/articles/rbioapi_do_enrich.html

2. If you need to specifically use PANTHER tools, you may do something along this line: 
I. Perform enrichment analysis using panther. 
II. Retrieve the genes in each GO term using [biomaRt](https://bioconductor.org/packages/release/bioc/html/biomaRt.html).
III. Do a simple set intersection with your input genes. 

I wouldn't recommend the second approach since there can be version differences, and you may fall into reproducibility issues. Nevertheless, If you need to, I can write a code snippet to do this step.
 
p.s: I have checked the Panther API references to make sure that this issue is not related to rbioapi implementation of their services. rbioapi is a bit behind as 3 endpoints are not implemented. However, this was not related to this issue.

**Comment by moosa-r** [comment]
> thanks for this helpful tool! after running rba_panther_enrich(), how can I get the list of genes in my gene list that are associated with each significant term? I can't seem to find a function that will allow that. Thanks!

Dear Tyssowski 

Unfortunately, this PANTHER API endpoint doesn't return the user's gene within each significant term. However, In case you are performing an over-representation analysis of your genes against GO terms, I think you have two options here: 
 
1. In rbioapi, use other enrichment services. Please see this article: 
https://rbioapi.moosa-r.com/articles/rbioapi_do_enrich.html

2. If you need to specifically use PANTHER tools, you may do something along this line: 
I. Perform enrichment analysis using panther. 
II. Retrieve the genes in each GO term using [biomaRt](https://bioconductor.org/packages/release/bioc/html/biomaRt.html).
III. Do a simple set intersection with your input genes. 

I wouldn't recommend the second approach since there can be version differences, and you may fall into reproducibility issues. Nevertheless, If you need to, I can write a code snippet to do this step.
 
p.s: I have checked the Panther API references to make sure that this issue is not related to rbioapi implementation of their services. rbioapi is a bit behind as 3 endpoints are not implemented. However, this was not related to this issue.

**Comment by erzakiev** [comment]
This is also interesting to me. I wonder if the Panther API still doesn't provide the genes that were 'hits' in the enriched terms?

And I'd prefer to still use Panther and its 'GO slims' databases as they seem to provide very eloquent (i.e. only a few lines) lists of terms instead of those lists of GO terms as provided, for example, by a simple over-representation search through the C2 collection of MSigDb using clusterProfiler.

**Comment by moosa-r** [comment]
Dear @erzakiev 
I see your point. The response format has"t changed. Adding an extra column to the response with genes hits is a trivial task. But to prevent future problems and keep the package maintainable, I need to be utterly faithful to the response format of the supported API.

If you need to use PANTHER via rbioapi, You can use something like the workaround I proposed above. But please keep in mind an important caveat: make sure to use the same release that PANTHER uses to map the GO terms to your genes.

**Comment by moosa-r** [comment]
Hi @ktyssowski and @erzakiev,

A long-overdue update on this request: when the issue was originally raised, the PANTHER API did not return the genes mapped to each enrichment term. It does now, and I’ve added support for this in rbioapi.

`rba_panther_enrich()` now has a `request_mapped_genes` argument, which can be set to `"input"` (the default), `"reference"`, or `"none"`. With the default setting, the input genes mapped to each term are returned in the `input_list.mapped_ids` column of `result`.

This change has not yet been submitted to CRAN, so it is not included in the current CRAN release (v0.8.3). For now, you can install the latest development version directly from GitHub:

```r
install.packages("remotes")  # if not already installed
remotes::install_github("moosa-r/rbioapi")
```

Best,
Moosa
